src/control/client/pool.go

```diff
@@ -27,30 +27,100 @@ import (
 	"golang.org/x/net/context"
 
 	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/pkg/errors"
 )
 
-// CreatePool will create a DAOS pool using provided parameters and return uuid
-func (c *connList) CreatePool(req *pb.CreatePoolReq) ResultMap {
-	results := make(ResultMap)
-	mc := c.controllers[0] // connect to first AP only for now
+func verifyUnicast(cs []Control) (Control, error) {
+	if len(cs) > 1 {
+		return nil, errors.Errorf("unexpected number of connections, "+
+			"want 1, have %d", len(cs))
+	}
 
-	resp, err := mc.getSvcClient().CreatePool(context.Background(), req)
+	return cs[0], nil
+}
 
-	result := ClientResult{mc.getAddress(), resp, err}
-	results[result.Address] = result
+// PoolCreateReq struct contains request
+type PoolCreateReq struct {
+	ScmBytes   uint64
+	NvmeBytes  uint64
+	RankList   string
+	NumSvcReps uint32
+	Sys        string
+	Usr        string
+	Grp        string
+}
 
-	return results
+// PoolCreateResp struct contains response
+type PoolCreateResp struct {
+	Uuid    string
+	SvcReps string
 }
 
-// DestroyPool will Destroy a DAOS pool identified by its UUID.
-func (c *connList) DestroyPool(req *pb.DestroyPoolReq) ResultMap {
-	results := make(ResultMap)
-	mc := c.controllers[0] // connect to first AP only for now
+// PoolCreate will create a DAOS pool using provided parameters and return
+// uuid, list of service replicas and error (including any DER code from DAOS).
+//
+// Isolate protobuf encapsulation in client and don't expose to calling code.
+func (c *connList) PoolCreate(log logging.Logger, req *PoolCreateReq) (
+	*PoolCreateResp, error) {
+
+	mc, err := verifyUnicast(c.controllers)
+	if err != nil {
+		return nil, err
+	}
+
+	rpcReq := &pb.PoolCreateReq{
+		Scmbytes: req.ScmBytes, Nvmebytes: req.NvmeBytes,
+		Ranks: req.RankList, Numsvcreps: req.NumSvcReps, Sys: req.Sys,
+		User: req.Usr, Usergroup: req.Grp,
+	}
+
+	log.Infof("Creating DAOS pool: %s\n", rpcReq)
```

Comment:
As noted before, connList has its own log member, so this should be `c.log.Infof()` and that will remove the need to pass it in as an argument.

Reply:
That having been noted, we really shouldn't be logging user-visible messages from within library code. I realize that there is currently a fair amount of precedent for this, but it doesn't mean that we should perpetuate it with new code. The place for user-visible messages is on the outside of these API boundaries. I wouldn't object to this being a Debug-level message, though.